src/IdentityServer/Validation/Models/ValidatedAuthorizeRequest.cs

```diff
@@ -138,26 +138,66 @@ public class ValidatedAuthorizeRequest : ValidatedRequest
     /// <summary>
     /// Gets or sets the collection of prompt modes.
     /// </summary>
+    /// <remarks>
+    /// The <see cref="PromptModes"/> change as they are used. For example, if
+    /// the prompt mode is login (to force the login UI to be displayed), the
+    /// collection will initially contain login, but when the login page is
+    /// displayed, the login prompt will be removed from the collection of
+    /// prompt modes so that the login page will only be displayed once.
+    /// <para>
+    /// See also: <see cref="SuppressedPromptModes"/> and <see
+    /// cref="OriginalPromptModes"/>.
+    /// </para>
+    /// </remarks>
     /// <value>
-    /// The collection of prompt modes.
+    /// The collection of prompt modes, which changes as the request is
+    /// processed and various prompts are displayed.
     /// </value>
     public IEnumerable<string> PromptModes { get; set; } = Enumerable.Empty<string>();
 
     /// <summary>
     /// Gets or sets the collection of original prompt modes.
     /// </summary>
+    /// <remarks>
+    /// The <see cref="PromptModes"/> change as they are used. For example, if
+    /// the prompt mode is login (to force the login UI to be displayed), the
+    /// collection will initially contain login, but when the login page is
+    /// displayed, the login prompt will be removed from the collection of
+    /// prompt modes so that the login page will only be displayed once.
+    /// <para>
+    /// See also:
+    /// <list type="bullet">
+    /// <item><seealso cref="SuppressedPromptModes"/></item>
+    /// <item><seealso cref="PromptModes"/></item>
+    /// </list>
+    /// </para>
+    /// </remarks>
     /// <value>
     /// The collection of original prompt modes.
     /// </value>
-    internal IEnumerable<string> OriginalPromptModes { get; set; } = Enumerable.Empty<string>();
+    public IEnumerable<string> OriginalPromptModes { get; set; } = Enumerable.Empty<string>();
 
     /// <summary>
     /// Gets or sets the collection of suppressed prompt modes.
     /// </summary>
+    /// <remarks>
+    /// The <see cref="PromptModes"/> change as they are used. For example, if
+    /// the prompt mode is login (to force the login UI to be displayed), the
+    /// collection will initially contain login, but when the login page is
+    /// displayed, the login prompt will be removed from the collection of
+    /// prompt modes so that the login page will only be displayed once.
+    /// </remarks>
+    /// <para>
+    /// See also:
+    /// <list type="bullet">
+    /// <item><seealso cref="PromptModes"/></item>
+    /// <item><seealso cref="OriginalPromptModes"/></item>
+    /// </list>
+    /// </para>
     /// <value>
     /// The collection of suppressed prompt modes.
     /// </value>
-    internal IEnumerable<string> SuppressedPromptModes { get; set; } = Enumerable.Empty<string>();
+    public IEnumerable<string> SuppressedPromptModes { get; set; } = Enumerable.Empty<string>();
```

Comment:
Maybe we want a better name for these, like "ProcessedPromptModes"? Thoughts? Suggestions?

Reply:
Ok, let's revert the constant and keep the new "Processed" name on the new prop, please.